packages/opentelemetry-scope-zone-peer-dep/src/ZoneScopeManager.ts

```diff
@@ -0,0 +1,258 @@
+        target,
+        target.addEventListener,
+        scope
+      );
+    }
+
+    if (typeof target.removeEventListener === 'function') {
+      target.removeEventListener = this._patchRemoveEventListener(
+        target,
+        target.removeEventListener
+      );
+    }
+
+    return obj;
+  }
+
+  /**
+   * Creates a new unique zone name
+   */
+  private _createZoneName() {
+    this._zoneCounter++;
+    const random = Math.random();
+    return `${this._zoneCounter}-${random}`;
+  }
+
+  /**
+   * Creates a new zone
+   * @param zoneName zone name
+   * @param scope A scope (span) to be bind with Zone
+   */
+  private _createZone(zoneName: string, scope: unknown): Zone {
+    return Zone.root.fork({
+      name: zoneName,
+      properties: {
+        [ZONE_SCOPE_KEY]: scope,
+      },
+    });
+  }
+
+  /**
+   * Returns the active zone
+   */
+  private _getActiveZone(): Zone | undefined {
+    return Zone.current;
+  }
+
+  /**
+   * Patches addEventListener method
+   * @param target any target that has "addEventListener" method
+   * @param original reference to the patched method
+   * @param [scope] scope to be bind to the listener
+   */
+  private _patchAddEventListener(
+    target: TargetWithEvents,
+    original: Function,
+    scope?: unknown
+  ) {
+    const scopeManager = this;
+
+    return function(this: {}, event: string, listener: Func<void>, opts?: any) {
+      if (target.__ot_listeners === undefined) {
+        target.__ot_listeners = {};
+      }
+      let listeners = target.__ot_listeners[event];
+      if (listeners === undefined) {
+        listeners = new WeakMap();
+        target.__ot_listeners[event] = listeners;
+      }
+      const patchedListener = scopeManager.bind(listener, scope);
+      // store a weak reference of the user listener to ours
+      listeners.set(listener, patchedListener);
+      return original.call(this, event, patchedListener, opts);
+    };
+  }
+
+  /**
+   * Patches removeEventListener method
+   * @param target any target that has "removeEventListener" method
+   * @param original reference to the patched method
+   */
+  private _patchRemoveEventListener(
+    target: TargetWithEvents,
+    original: Function
+  ) {
+    return function(this: {}, event: string, listener: Func<void>) {
+      if (
+        target.__ot_listeners === undefined ||
+        target.__ot_listeners[event] === undefined
+      ) {
+        return original.call(this, event, listener);
+      }
+      const events = target.__ot_listeners[event];
+      const patchedListener = events.get(listener);
+      events.delete(listener);
+      return original.call(this, event, patchedListener || listener);
+    };
+  }
+
+  /**
+   * Returns the active scope
+   */
+  active(): unknown | undefined {
+    const activeZone = this._getActiveZone();
+
+    const active = this._activeScopeFromZone(activeZone);
+    if (active) {
+      return active;
+    }
+    if (this._enabled) {
+      return window;
```

Comment:
This is just adding extra complexity. Also the Zone is global object, Zone.root will be shared among all instances of web tracers. But managing the scopes is the responsibility of `Scope Zone Manager`. I think that pushing the extra information to object that many instances will be used is just asking for more trouble in future. What if for any reason you will want to remove it. Which web tracer instance should do it then ?.

Reply:
I guess I think of it as `Zone.root` and `window` as equally shared among all web tracers. This is more of a style thing I think for me - just having a consistent place to store the scope, where it always is in `Zone.current.get('...`. It would of course be globally shared if its the "background" scope.
Does that help clarify?

I'm open if you want to keep it as `window`

Reply:
I mean that because Zone root is a global by default then anyone or anything can change that - not only web tracers, maybe even a different version of open-telemetry might use it differently. So having that in mind can we guarantee that once it is set, it will stay as it is forever ?. Once we create a zone then we have full control over it, but for the zone that already exists like a root, not really. And also some web tracers might be disabled. If tracer is disabled then `active` is returning `undefined`, not a window. That's why I would say I can't rely on Zone root always. That's why I want to avoid it by keeping this responsibility in `scope manager`